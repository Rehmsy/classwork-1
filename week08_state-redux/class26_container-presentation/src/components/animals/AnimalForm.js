import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AnimalForm extends Component {

  state = { 
    editing: false,
    name: '',
    type: ''
  };

  static propTypes = {
    onComplete: PropTypes.func.isRequired
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { name, type } = this.state;
    this.props.onComplete({ name, type });
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  render() { 
    const { name, type } = this.state;

    return (
      <form onSubmit={this.handleSubmit}>
        <InputControl name="name" value={name} onChange={this.handleChange}/>
        <InputControl name="type" value={type} onChange={this.handleChange}/>
        <p>
          <button>Add</button>
        </p>
      </form>
    );
  }
}

const InputControl = (props) => (
  <p>
    <label>
      {props.name}:
      <input {...props} required/>
    </label>
  </p>
);
 
export default AnimalForm;