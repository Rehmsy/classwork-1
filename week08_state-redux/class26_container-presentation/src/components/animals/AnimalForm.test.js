import AnimalForm from './AnimalForm';
import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Animal Form', () => {
  
  it('renders add if no animal prop', () => {
    const handleComplete = jest.fn();

    const wrapper = mount(<AnimalForm onComplete={handleComplete}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();

    const animal = {
      name: 'felix',
      type: 'cat'
    };

    wrapper.find('input[name="name"]').simulate('change', {
      target: { 
        name: 'name',
        value: animal.name
      }
    });
    
    wrapper.find('input[name="type"]').simulate('change', {
      target: { 
        name: 'type',
        value: animal.type 
      }
    });

    wrapper.find('button').simulate('submit');

    // test we got the search term
    const calls = handleComplete.mock.calls;
    expect(calls.length).toBe(1); // only called once
    expect(calls[0][0]).toEqual(animal);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
