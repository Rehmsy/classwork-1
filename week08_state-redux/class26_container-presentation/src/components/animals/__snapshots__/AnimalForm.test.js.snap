// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Form renders add if no animal prop 1`] = `
<AnimalForm
  onComplete={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <InputControl
      name="name"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          name
          :
          <input
            name="name"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="type"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          type
          :
          <input
            name="type"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button>
        Add
      </button>
    </p>
  </form>
</AnimalForm>
`;

exports[`Animal Form renders add if no animal prop 2`] = `
<AnimalForm
  onComplete={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "name": "felix",
            "type": "cat",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <InputControl
      name="name"
      onChange={[Function]}
      value="felix"
    >
      <p>
        <label>
          name
          :
          <input
            name="name"
            onChange={[Function]}
            required={true}
            value="felix"
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="type"
      onChange={[Function]}
      value="cat"
    >
      <p>
        <label>
          type
          :
          <input
            name="type"
            onChange={[Function]}
            required={true}
            value="cat"
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button>
        Add
      </button>
    </p>
  </form>
</AnimalForm>
`;
